### PR TITLE
Add pika 0.2.0

### PR DESCRIPTION
**Pull request by msimberg**
Also adds an `apex` variant which is available from 0.2.0 onwards.

### CODE REVIEW

File: var/spack/repos/builtin/packages/pika/package.py [hunk 1]
```diff
@@ -44,6 +45,7 @@ class Pika(CMakePackage, CudaPackage, ROCmPackage):
 
     variant('examples', default=False, description='Build and install examples')
     variant('mpi', default=False, description='Enable MPI support')
+    variant('apex', default=False, description='Enable APEX support')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/pika/package.py]:
```suggestion
    variant('apex', default=False, description='Enable APEX support', when='@0.2.0:')
```

File: var/spack/repos/builtin/packages/pika/package.py [hunk 2]
```diff
@@ -63,6 +65,8 @@ class Pika(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on('mpi', when='+mpi')
     depends_on('cuda@11:', when='+cuda')
+    depends_on('apex', when='+apex')
+    conflicts('+apex', when='@:0.1')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/pika/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/pika/package.py [hunk 3]
```diff
@@ -89,6 +93,7 @@ def cmake_args(self):
             self.define_from_variant('PIKA_WITH_CUDA', 'cuda'),
             self.define_from_variant('PIKA_WITH_HIP', 'rocm'),
             self.define_from_variant('PIKA_WITH_MPI', 'mpi'),
+            self.define_from_variant('PIKA_WITH_APEX', 'apex'),
```
Comment (haampie) [inline var/spack/repos/builtin/packages/pika/package.py:95]: i think this is okay also for 0.1.0 (https://github.com/spack/spack/pull/27503)

**Review by haampie — APPROVED** [review]

**Review by albestro — APPROVED** [review]

**Review by teonnik — APPROVED** [review]

**Comment by haampie** [comment]
@scottwittenburg gitlab :heavy_check_mark: does not appear on github blocking automerge...

**Comment by haampie** [comment]
@spackbot run pipeline

**Pull request merged**